### PR TITLE
Allow overriding risk magnitudes

### PR DESCRIPTION
**Pull request by jaym**
Allows overriding the risk magnitude for a risk and completely disabling a risk.

After #1352

### CODE REVIEW

**Comment by jaym** [comment]
this is not ready yet. We need to pick the correct checks for the risk factor

File: policy/cnspec_policy.proto [hunk 1]
```diff
@@ -192,8 +197,9 @@ message RiskFactor {
   repeated cnquery.explorer.Mquery checks = 2;
 
   ScopeType scope = 70;
-  float magnitude = 71;
-  bool is_absolute = 72;
+  float deprecated_v11_magnitude = 71;
+  bool deprecated_v11_is_absolute = 72;
```
Comment (czunker) [inline policy/cnspec_policy.proto]: nit: Could you please add a comment what to use instead?

File: policy/cnspec_policy.proto [hunk 2]
```diff
@@ -645,7 +653,7 @@ message ScoreDelta {
 message ScoredRiskFactor {
   string mrn = 1;
   float risk = 2;
-  bool is_absolute = 3;
```
Comment (czunker) [inline policy/cnspec_policy.proto]: I know, we save a risk factor struct in the backend. Do you know whether it is this one? Would this break the unmarshal of already saved risk factors?

File: policy/resolver.go [hunk 3]
```diff
@@ -710,6 +719,42 @@ func (s *LocalServices) refreshChecksums(executionJob *ExecutionJob, collectorJo
...
+}
+
+func (s *LocalServices) mergeRisk(cache *resolverCache, riskFactor *RiskFactor) {
+	if existing, ok := cache.riskFactors[riskFactor.Mrn]; ok {
+		if riskFactor.Magnitude != nil {
+			existing.Magnitude = riskFactor.Magnitude
+		}
+
+		if riskFactor.Action != explorer.Action_UNSPECIFIED {
+			existing.Action = riskFactor.Action
```
Comment (czunker) [inline policy/resolver.go:750]: Perhaps an edge case:
Could the same risk factor be defined in multiple policies?
This line would then mean that the last non-unspecified action wins. Is this as it should be?

Comment (jaym) [reply policy/resolver.go:750]: ya, i think that problem already exists. But if its defined in the space policy it will win

**Review by czunker — APPROVED** [review]

**Pull request merged**